pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -383,10 +383,20 @@ func (gb *GraphBuilder) addDependentToOwners(n *node, owners []metav1.OwnerRefer
 			// attemptToDelete. The garbage processor will enqueue a virtual delete
 			// event to delete it from the graph if API server confirms this
 			// owner doesn't exist.
+			fqKind := schema.FromAPIVersionAndKind(owner.APIVersion, owner.Kind)
+			mapping, err := gb.restMapper.RESTMapping(fqKind.GroupKind(), owner.APIVersion)
+			// TODO: This assumption won't hold if the owner lives in another namespace.
+			ownerNamespace := n.identity.Namespace
+			if err != nil {
+				// TODO: How should failures be handled? Requeue seems appropriate.
+				glog.V(0).Infof("restmapping failed for owner: %v: %v", owner, err)
```

Comment:
@caesarxuchao think we should change the signature of this function to return an error and allow re-processing? Not entirely sure yet what is the best way to actually cause a retry.

Reply:
How about this: we don't check the restmapper when adding the virtual node. When the gc process the virtual node, it calls [gc.getObject](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/garbagecollector.go#L384). The [getObject](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/operations.go#L72) function uses restmapper already, we use that restmapper to double check if the object is namespace scoped.

The benefit is that if the restmapper is stale, a restmapper error is returned, the object will be automatically added back to the `attemptToDelete` queue and will be retried.

Reply:
The discussion in https://github.com/kubernetes/kubernetes/issues/54940 seems to be implying that the fact cluster scoped owners even works at all today is accidental (given the concerns raised about security/policy). Am I misunderstanding? The API change seems preferable to reinforcing the implicit behavior we have now.

Reply:
Right, that was not in the original design.

What specifically are the security/policy concerns regarding having cluster-scoped owners?